plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -42,6 +44,29 @@ export class TeamManager {
         }
     }
 
+    async flushLastSeenAtCache(): Promise<void> {
+        const team_ids: string[] = []
+        const event_names: string[] = []
+        const last_seen_at_array: number[] = []
+
+        for (const event of this.eventLastSeenCache) {
```

Comment:
I think this is not safe for concurrent modifications, which will happen here. You want to do:

```
const events = this.eventLastSeenCache
this.eventLastSeenCache = new Map()
```

just above here.